### ISSUE TITLE
[bug] can't save videos

### ISSUE THREAD

**Issue by TobiPeterG** [Docile-Alligator/Infinity-For-Reddit#690]
Hey there,

I recently noticed that I can't save videos anymore with the infinity app, it tells me that it has no access to the temporary folder.

I'm on Android 12, MIUI 13, using latest beta1
Can this be confirmed?

**Comment by Wladefant** [comment]
Is it still an issue?

**Comment by TobiPeterG** [comment]
> @TobiPeterG Is it still an issue?

Yes :(

It tells me that the temporary folder is not accessible

**Comment by schqual** [comment]
Confirm this issue. But my config is different:
Android 13, Mi10, AOSP Rom (Picel Experience Unofficial)

**Comment by dam-cav** [comment]
Hi, I also encountered this problem last week after changing rom, I think I found a workaround that explains the cause of the problem
I can tell you WHY it happens, but not HOW to fix directly in the app (I'm not a super-expert of android native code, hope someone is able to come with a solution thanks to these informations)

I recently switched from the ArrowOS ROM, where the problem wasn't there, to CrDroid ROM, where the problem IS there, both are on Android 13 (the device is Xiaomi Davinci but I don't think it's an important detail)

I noticed that the two roms manage differently the file permissions of the applications located in the `/data/data` folder

In the case of ArrowOS (where video download works), application files have permissions `700` and all files belong to the user and group `u0_aXYZ` (random user generated when the app is installed)

In the case of CrDroid, files have `771` permissions by default, and in particular cache folders belong to a different group `u0_aXYZ_cache` and have `rwxrws--x` permissions (I googled that `s` in position 6 and it stands for `setuid`)

Noticed this, the way I applied the workaround is as follow:

1. reboot into recovery (Or maybe do the same as the next steps from terminal emulator with root, IDK, I'm using TWRP)
3. run `adb shell`
4. `cd /data/data/ml.docilealligator.infinityforreddit` to reach app directory, find the `XYZ` number in `u0_aXYZ` autoassigned to app user
5. change user group of cache folders back to the same group of other folders with `chown -R u0_aXYZ:u0_aXYZ cache` and `chown -R u0_aXYZ:u0_aXYZ code_cache`
6. rebooted, and now I'm able to download video again

In short it seems that the cache folders in some roms have different permissions and that the app is not using the correct user to access the files in these cases (?)

Image for context:
![infinity_workaround](https://github.com/Docile-Alligator/Infinity-For-Reddit/assets/54503043/bdb68a27-b335-4dc3-9eb8-44ea1b384d21)

**Comment by Simon1511** [comment]
Still having this issue in 2024 on an A52s with Android 14. Changing the download folder doesn't change anything.
